compiler/hash-typecheck/src/ops/substitute.rs

```diff
@@ -573,6 +579,27 @@ impl<'gs, 'ls, 'cd, 's> Substituter<'gs, 'ls, 'cd, 's> {
         self.add_free_vars_in_term_to_set(term_id, &mut result);
         result
     }
+
+    /// Get the set of variables that exist in the given term. This function
+    /// will proceed an error if it hits an un-resolved free variable since this
+    /// cannot appear within certain contexts, e.g. nominal definitions.
+    pub fn get_vars_in_term(&self, term_id: TermId) -> TcResult<HashSet<Var>> {
+        let mut result = HashSet::new();
+        self.add_free_vars_in_term_to_set(term_id, &mut result);
+
+        result
+            .into_iter()
+            .map(|var| match var {
+                SubSubject::Var(var) => Ok(var),
+                SubSubject::Unresolved(_) => Err(TcError::UnresolvedVariable {
+                    // @@Correctness: it's unclear if we can even get an identifier here? Or if we
+                    // should make an identifier from the provided `ResolutionId`
```

Comment:
You can make a string by `self.for_fmt(var.into())` and then create an identifier from it.

Reply:
There is no implementation for `UnresolvedTerm`, should I implement one in this pr?